core_types/src/chain_header.rs

```diff
@@ -109,25 +114,22 @@ impl ToEntry for ChainHeader {
     fn to_entry(&self) -> Entry {
         Entry::new(
             &EntryType::ChainHeader,
-            &self.to_json().expect("entry should be valid"),
+            &JsonString::from(self.to_owned()),
         )
     }
 
     fn from_entry(entry: &Entry) -> Self {
-        return ChainHeader::from_json_str(&entry.value())
-            .expect("entry is not a valid ChainHeader Entry");
+        ChainHeader::from(entry.value().to_owned())
     }
 }
 
 impl AddressableContent for ChainHeader {
     fn content(&self) -> Content {
-        self.to_json()
-            .expect("could not Jsonify ChainHeader as Content")
+        Content::from(self.to_owned())
     }
 
     fn from_content(content: &Content) -> Self {
-        ChainHeader::from_json_str(content)
-            .expect("could not read Json as valid ChainHeader Content")
+        Self::from(content.to_owned())
```

Comment:
Shouldn't `ChainHeader::from_json_str()` be removed from codebase?

Reply:
probably yes